src/components/Dashboard/FilingHistoryList.vue

```diff
@@ -1092,6 +1109,47 @@ export default {
 
       return (this.disableChanges || item.isCorrection || item.isFutureEffectiveIa || item.isColinFiling ||
         disableThisIaCorrection)
+    },
+
+    /**
+     * Fetches a filing by its id.
+     * @param businessId The business identifier
+     * @param filingId The filing identifier
+     * @returns a promise to return the filing of the specified type
+     */
+    fetchFilingById (businessId: string, filingId: number): Promise<any> {
+      const url = `businesses/${businessId}/filings/${filingId}`
+      return axios.get(url)
+        .then(response => {
+          if (response && response.data) {
+            return response.data.filing
+          } else {
+            // eslint-disable-next-line no-console
+            console.log('fetchFilingById() error - invalid response =', response)
+            throw new Error('Invalid API response')
+          }
+        })
+    },
+
+    /**
+     * Create a draft correction filing.
+     * @param businessId The business identifier
+     * @param filing the object body of the request
+     * @returns a promise to return the filing
+     */
+    createCorrection (businessId: string, filing: CorrectionFilingIF): Promise<any> {
```

Comment:
I think we should get in the habit of putting these calls in a mixin (or service) module.

I suggest generalizing the name ("create filing?") and the filing type can be "any", so that this can be reused.

Reply:
For sure! As mentioned above, we don't have a mixin handling requests in the Filings UI. Everything is pretty contained in the components but can certainly create one!

Reply:
Yes please!